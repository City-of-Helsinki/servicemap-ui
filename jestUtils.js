import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import english from './src/i18n/en';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'en',
  messages: english,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });
