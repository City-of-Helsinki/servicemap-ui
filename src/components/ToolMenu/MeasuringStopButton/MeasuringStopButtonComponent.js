import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import SMButton from '../../ServiceMapButton';


const MeasuringStopButtonComponent = ({
  classes,
  onClick,
}) => {
  const buttonRef = useRef();
  useEffect(() => {
    // When Measuring is activated focus should move to measuring close button
    if (buttonRef?.current) {
      buttonRef.current.focus();
    }
  }, []);

  return (
    <SMButton
      passingRef={buttonRef}
      id="MeasuringStopButton"
      className={classes.measuringButton}
      color="primary"
      role="button"
      messageID="tool.measuring.stop"
      onClick={onClick}
    />
  );
};


MeasuringStopButtonComponent.propTypes = {
  classes: PropTypes.shape({
    menuContainer: PropTypes.string,
    smIcon: PropTypes.string,
    measuringButton: PropTypes.string,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MeasuringStopButtonComponent;
