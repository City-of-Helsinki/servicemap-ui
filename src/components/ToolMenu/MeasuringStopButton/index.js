import { withStyles } from '@mui/styles';
import MeasuringStopButtonComponent from './MeasuringStopButtonComponent';
import styles from '../styles';

export default withStyles(styles)(MeasuringStopButtonComponent);
