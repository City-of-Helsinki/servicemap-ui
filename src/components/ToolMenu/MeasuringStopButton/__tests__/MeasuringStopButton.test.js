// Link.react.test.js
import React from 'react';
import { fireEvent } from '@testing-library/react';
import MeasuringStopButton from '../index';
import { getRenderWithProviders } from '../../../../../jestUtils';
import englishTranslations from '../../../../i18n/en';


const renderWithProviders = getRenderWithProviders({});

describe('<MeasuringStopButton />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(
      <MeasuringStopButton onClick={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const { getByRole } = renderWithProviders(
      <MeasuringStopButton
        onClick={mockCallBack}
      />,
    );
    fireEvent.click(getByRole('button'));
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('does render accessibility attributes correctly', () => {
    const { getByRole } = renderWithProviders(
      <MeasuringStopButton />,
    );
    const buttonBase = getByRole('button');
    const p = buttonBase.querySelector('p');
    const contentText = englishTranslations['tool.measuring.stop'];
    // Expect aria-label to be same as text content
    expect(buttonBase).toHaveAttribute('aria-label', contentText);
    // Expect aria-hidden to be false Eficode report 05-2022 page 47
    expect(buttonBase.getAttribute('aria-hidden')).toBeFalsy();
    // Expect visible text to be hidden from screen readers
    expect(p).toHaveAttribute('aria-hidden', 'true');
    // Expect visible text to be same as aria-label
    expect(p).toHaveTextContent(contentText);
  });
});
