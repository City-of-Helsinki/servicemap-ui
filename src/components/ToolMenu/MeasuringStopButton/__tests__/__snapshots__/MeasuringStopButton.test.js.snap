// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasuringStopButton /> should work 1`] = `
<div>
  <button
    aria-label="Stop measuring"
    class="MuiButtonBase-root SMButton SMButton-button-7  SMButton-marginRight-10 MeasuringStopButtonComponent-measuringButton-6 SMButton-primary-11 css-fvdnvy-MuiButtonBase-root"
    id="MeasuringStopButton"
    role="button"
    tabindex="0"
    type="button"
    variant="contained"
  >
    <p
      aria-hidden="true"
      class="MuiTypography-root MuiTypography-caption SMButton-typography-14 css-o763o0-MuiTypography-root"
    >
      Stop measuring
    </p>
  </button>
</div>
`;
